include/alpaka/meta/SyclPrintf.hpp

```diff
@@ -0,0 +1,35 @@
+/* Copyright 2022 Andrea Bocci
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <CL/sycl.hpp>
+
+#ifdef ALPAKA_ACC_SYCL_ENABLED
+
+// Kill printf in AMD GPU code because of missing compiler support
+#    ifdef __AMDGCN__
+#        include <cstdio> // the define breaks <cstdio> if it is included afterwards
+#        define printf(...)
+#    else
+
+#        ifdef __SYCL_DEVICE_ONLY__
+#            define CONSTANT __attribute__((opencl_constant))
```

Comment:
```suggestion
#            define CONSTANT [[clang::opencl_constant]]
```

Reply:
With this change I get a compiler error on the `printf`:
```bash
error: an attribute list cannot appear here
```
Can you tell me why you didn't want `__attribute__((opencl_constant))`?

Reply:
I just prefer the C++ attribute syntax. It doesn't look as ugly :-)